### ISSUE TITLE
[2.5.3] SQL queries executed using the JPA API are not logged by jdbcdslog

### ISSUE THREAD

**Issue by sevasm** [playframework/playframework#6149]
### Play Version (2.5.x / etc)

2.5.3

### API (Scala / Java / Neither / Both)

Java

### Operating System (Ubuntu 15.10 / MacOS 10.10 / Windows 10)

Windows 8.1

### JDK (Oracle 1.8.0_72, OpenJDK 1.8.x, Azul Zing)

java version "1.8.0_40"
Java(TM) SE Runtime Environment (build 1.8.0_40-b26)
Java HotSpot(TM) 64-Bit Server VM (build 25.40-b25, mixed mode)

### Library Dependencies

N/A

### Expected Behavior

Please describe the expected behavior of the issue, starting from the first action.

1. Expose database connection to JNDI with default.jndiName = "DefaultDS".
2. Add and configure JPA implementation (I use Hibernate 4.3.11.Final).
3. Declare default JPA persistence unit with jpa.default = defaultPersistenceUnit
4. Configure JPA implementation to use the same data source and persistence unit names.
5. Enable SQL query logging as per https://www.playframework.com/documentation/2.5.x/JavaDatabase#How-to-configure-SQL-log-statement
6. Create and wire a controller with a method that executes at least one query using the JPA API (preferably via an injected play.db.jpa.JPAApi instance).
7. Access the controller method.
8. Inspect the console output or application logs. The SQL queries that were executed should be included in both.

### Actual Behavior

No SQL queries executed via the JPA API are logged.

### Reproducible Test Case

A basic sample application that illustrates the issue can be found here:

https://github.com/sevasm/play-sql-log

**Comment by sevasm** [comment]
I have submitted this issue also to the Play Google group here:

https://groups.google.com/d/msgid/play-framework/b8628fd0-8a84-48f9-a2fc-9ac6344b8f17%40googlegroups.com?utm_medium=email&utm_source=footer

My investigation of the problem leads me to believe that the issue is caused by the fact that jdbcslog is configured at the Database API level and not at the data source level (copy of text from Play Google group post follows):

Looking at the source for Play 2.5.3 I realised that only the data source used by the Database API is wrapped to enable SQL logging, see this excerpt from Databases.scala:

```
def createDataSource(): DataSource = {
  val datasource: DataSource = pool.create(name, databaseConfig, configuration)
  if (configuration.getBoolean("logSql")) {
    val proxyDatasource = new LogSqlDataSource()
    proxyDatasource.setTargetDSDirect(datasource)
    proxyDatasource
  } else {
    datasource
  }
}
```

As opposed to this from HikariCPModule where an unwrapped data source is bound in JNDI:

```
val hikariConfig = new HikariCPConfig(dbConfig, config).toHikariConfig
val datasource = new HikariDataSource(hikariConfig)

// Bind in JNDI
dbConfig.jndiName.foreach { jndiName =>
  JNDI.initialContext.rebind(jndiName, datasource)
  logger.info(s"datasource [$name] bound to JNDI as $jndiName")
}
```

As a proof-of-concept I was able to get SQL logging to work with the JPA API by modifying and building Play to wrap the data source when creating it in HikariCPModule.scala (copy of text from Play Google group post follows):

I modified and built Play locally after making the following changes to HikariCPModule.scala:

Changed this:

`val datasource = new HikariDataSource(hikariConfig)`

to this:

```
val originalDatasource = new HikariDataSource(hikariConfig)
val datasource = new org.jdbcdslog.LogSqlDataSource()
datasource.setTargetDSDirect(originalDatasource)
```

The application was then able to log the SQL executed by Hibernate. Of course making this change alone is not enough:

1. The data source is now wrapped twice when using the database API (once at the connection pool level and again at the database API level).
2. it does not check whether db.default.logSql is set to true before wrapping the data source
3. I have not considered BoneCP when making the change, but looking at the code for BoneCPModule it looks like a similar approach should work.

However, the change illustrates that adding SQL logging at the connection pool level works.

I am wondering if a proper change in this direction would be a suitable candidate for a patch for the next version of Play 2.5?
